waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/HivePrefixPattern.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (C) 2016-2025 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+
+public class HivePrefixPattern {
+  private final String prefix;
+  private String pattern;
+  private final List<String> subPatterns;
+
+  @VisibleForTesting
+  static boolean[] matchDp(String input, String pattern) {
+    int m = input.length() + 1;
+    int n = pattern.length() + 1;
+    boolean[] dp = new boolean[n];
```

Comment:
dp?

Reply:
This method employs the dynamic programming algorithm. I have modified the variable names and added some comments.